### PR TITLE
feat(openai): add /openai/chatgpt/release-notes

### PR DESCRIPTION
**Pull request by ETiV**
## 该 PR 相关 Issue / Involved issue

Close #无

## 路由地址示例 / Example for the proposed route(s)

```routes
/openai/chatgpt/release-notes
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [ ] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/openai/chatgpt.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const dayjs = require('dayjs');
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+
```
Comment (TonyRL) [inline lib/v2/openai/chatgpt.js:4]:
```suggestion
const config = require('@/config').value;

```

File: lib/v2/openai/chatgpt.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return returns;
+    });
+
+    ctx.state.data = {
+        title: feedTitle,
+        description: feedDesc,
+        link: articleUrl,
+        item: items,
+    };
```
Comment (TonyRL) [inline lib/v2/openai/chatgpt.js:92]:
```suggestion
    const cache = await ctx.cache.tryGet(
        articleUrl,
        async () => {
            const returns = [];

            const pageResponse = await got({
                method: 'get',
                url: articleUrl,
            });

            const $ = cheerio.load(pageResponse.data);
            const page = JSON.parse($('script#__NEXT_DATA__').text()); // 页面貌似是用 Next 渲染的，有现成的 JSON 数据可以直接 parse

            const feedTitle = page.props.pageProps.articleContent.title;
            const feedDesc = page.props.pageProps.articleContent.description;
            const $author = page.props.pageProps.articleContent.author;
            const authorName = $author.name;

            const $blocks = page.props.pageProps.articleContent.blocks;

            const anchorDay = dayjs();
            let heading = null,
                articleObj = {};
            let year = anchorDay.year();
            let prevMonth = null;

            $blocks.forEach((block) => {
                const text = (block.text || '').trim();
                if (!text) {
                    return;
                }
                if (block.type === 'subheading') {
                    if (heading !== null) {
                        articleObj.description = articleObj.description.trim().replace(/\n/g, '<br/>');
                        returns.push(articleObj);
                        articleObj = {};
                    }

                    heading = text;

                    articleObj.title = heading;
                    articleObj.author = authorName;
                    articleObj.category = 'ChatGPT';
                    articleObj.link = articleUrl + '#' + block.idAttribute;
                    articleObj.description = '';

                    const matchesPubDate = heading.match(/\((\w+\s\d{1,2})\)$/);
                    // 目前 ChatGPT Release Notes 页面并没有写入年份，所以只能靠猜
                    // 实现：当年度交替时，年份减去 1
                    if (matchesPubDate !== null) {
                        const curMonth = matchesPubDate[1].split(' ')[0];
                        if (prevMonth === 'Jan' && prevMonth !== curMonth) {
                            year--; // 年度交替：上一个月份是 Jan，现在的不是 Jan
                        }

                        prevMonth = curMonth;
                        const pubDay = dayjs(`${matchesPubDate[1]}, ${year}`, ['MMMM D, YYYY', 'MMM D, YYYY'], 'en', true);
                        // 从 ISO（GMT）时间的字符串（使用字符串替换的方式）替换成 US/Pacific PST 时区的表达
                        articleObj.pubDate = dayjs(pubDay.toISOString().replace(/\.\d{3}Z$/, '-08:00'));

                        const linkAnchor = pubDay.format('YYYY_MM_DD');
                        articleObj.guid = articleUrl + '#' + linkAnchor;
                    }
                } else {
                    articleObj.description += block.text.trim() + '\n\n';
                }
            });

            if (heading !== null) {
                articleObj.description = articleObj.description.trim().replace(/\n/g, '<br/>');
                returns.push(articleObj);
            }

            return { feedTitle, feedDesc, items: returns };
        },
        config.cache.routeExpire,
        false
    );

    ctx.state.data = {
        title: cache.feedTitle,
        description: cache.feedDesc,
        link: articleUrl,
        item: cache.items,
    };
```

**Review by TonyRL — COMMENTED** [review]
- `feedTitle` and `feedDesc` will be `''` under a cache-hit situation.
- Used tighter regex `/\((\w+\s\d{1,2})\)$/` vs `/\((.+\d{1,2})\)$/`
- Since you're caching the list and their content, keep in mind that new items won't appear if you don't specify the 4th parameter as `false`

**Comment by ETiV** [comment]
我先把你建议的给合了，晚点再修一下你提到的 1、2 的问题

File: docs/new-media.md [hunk 3]
```diff
@@ -1032,6 +1032,10 @@ IPFS 网关有可能失效，那时候换成其他网关。
 
 </Route>
 
+### ChatGPT - Release Notes
+
+<Route author="ETiV" example="/openai/chatgpt/release-notes" path="/openai/chatgpt/release-notes" />
```
Comment (TonyRL) [inline docs/new-media.md:1037]: Missing en docs

**Pull request merged**